platform-hub-api/spec/controllers/users_controller_spec.rb

```diff
@@ -370,13 +366,66 @@
 
     it_behaves_like 'authenticated' do
 
+      let :git_hub_agent_service do
+        instance_double('Agents::GitHubAgentService')
+      end
+
       it_behaves_like 'not an admin so forbidden'  do
         before do
-          get :offboard_github, params: { id: @user.id }
+          post :offboard_github, params: { id: @user.id }
+        end
+      end
+
+      it_behaves_like 'an admin' do
+
+        context 'when user does not have a GitHub identity connected' do
+          it 'should return a 400 Bad Request with an appropriate error message' do
+            expect(Agents::GitHubAgentService).to receive(:new).with(any_args).and_return(git_hub_agent_service)
+            expect(git_hub_agent_service).to receive(:offboard_user).with(@user).and_raise(Agents::GitHubAgentService::Errors::IdentityMissing)
+            post :offboard_github, params: { id: @user.id }
+            expect(response).to have_http_status(400)
```

Comment:
I'm wondering if 422 would be more appropriate here?

Reply:
The problem I had with using [422](https://httpstatuses.com/422) here is it implies that the input data was wrong in some way. Whereas, it's really that the internal state of the user is not compatible with this request, so I'd like to say to the client that they shouldn't have even made this request in the first place (without checking that the user already has a GitHub identity connected).

400 is a bit of a catchall and blunt instrument here, but I'm not sure there's a better code we can use here?

Reply:
yeah. leave that with 400 for now.